### DISCUSSION TITLE
How to publish and host a HTML Quarto book in Microsoft Teams / Sharepoint?

### DISCUSSION THREAD

**Discussion by MohammedAliSharafuddin** [quarto-dev/quarto-cli#3535] [Q&A]
I rendered a HTML book using Quarto. I would like to host it privately in Microsoft Teams and Sharepoint for teaching purpose. I prefer updating the book on weekly basis by adding chapters. But I cannot find a straightforward solution to add the book into sharepoint. Because, sharepoint requirements are .aspx. The book is in HTML format. 
Is there any way to render a Quarto HTML book to Sharepoint from Rstudio?

**Comment by rgaiacs** [comment]
@MohamedaliS Thanks for your question. From a quick search, you must rename all `.html` files to `.aspx` files. You will have to change all the links in the books to point to the correct file.

You could zip the book and share the `.zip` file in Sharepoint.

Another option is to render your project to DOCX and share the DOCX in Sharepoint.

**Reply by MohammedAliSharafuddin** [reply]
@rgaiacs Thanks for the reply. 

These solutions wont work in my case, because

1) My book consists of more than 70 html files and renaming every file and link is a painful task. 
2) It is not possible to update contents automatically (Let's say, if I want to add some chapters / html pages, I have to do the renaming from the beginning).
3) I prefer web version. Because my book contains media contents (YouTube videos, Podcasts), link to Microsoft Quiz, and a lot more of other links. Docx is not a suitable solution.  

Just noticed the same issue with bookdown. https://github.com/rstudio/bookdown/issues/1366

**Reply by mcanouil** [reply]
See project script for such tedious task: https://quarto.org/docs/projects/scripts.html

**Comment by usrbinr** [comment]
Hi -- I've tried the suggested solution and I still can't properly render a quarto document in sharepoint.

Any other thoughts or suggestions besides renaming the extension. 

Note, I've had more luck rmarkdown with the below arguments. 

When I replicate these in quarto, I don't have success (eg. using self-contained or embed-resources)
output: 
      html_document:
        css: www/style.css   
        toc: true
        toc_float: true
        df_print: paged
        self_contained: yes
        mode: selfcontained

**Comment by allefeld** [comment]
You don't have to rename the files later, you can simply set the output file extension:

```yaml
format:
  html:
    output-ext: aspx
```

If you then upload the generated `aspx` files to SharePoint, including the subfolder containing external resources, it should simply work.

**Comment by jkkishore85** [comment]
I tried using aspx, but it didn't work for me. Instead, the following configuration in `_quarto.yml` worked fine:
`project:
  type: website

format:
  html:
    self-contained: true`

I also wrote a tutorial about this on my website: https://jkkishore85.github.io/tutorials/2024-07-17-quartotutorial-sharepoint-01/

**Reply by mcanouil** [reply]
Don't use `self-contained` use `embed-resources`.
The former has been superseded a long time again by Pandoc and Quarto.

Additionally, using this option on a Book or Website can lead to unexpected behaviour and can break some components that are shared across the pages.

`embed-resources` is supposed to be used for single file documents.
